engine/execution/provider/engine.go

```diff
@@ -185,15 +188,36 @@ func (e *Engine) onChunkDataRequest(
 		Collection:    collection,
 	}
 
-	// sends requested chunk data pack to the requester
-	err = e.chunksConduit.Unicast(response, originID)
-	if err != nil {
-		return fmt.Errorf("could not send requested chunk data pack to (%s): %w", origin, err)
+	sinceProcess := time.Since(processStart)
+
+	log.With().Dur("sinceProcess", sinceProcess).Logger()
+
+	if sinceProcess > time.Second*10 {
+		log.Warn().Msg("chunk data pack query takes longer than 10 secs")
 	}
 
-	log.Debug().
-		Hex("collection_id", logging.ID(response.Collection.ID())).
-		Msg("chunk data pack request successfully replied")
+	// sends requested chunk data pack to the requester
+	e.unit.Launch(func() {
+		deliveryStart := time.Now()
+
+		err = e.chunksConduit.Unicast(response, originID)
```

Comment:
maybe we can actually just use `:=` here, for clarity? might be confusing if we ever look at err outside this Launch, even though we don't right now.